_posts/2017-04-12-managing-technical-debt.md

```diff
@@ -0,0 +1,83 @@
+
+Without simple and understandable deployment process team will have a hard time on delivering the fixes and new features, and this impact a application users which is critical. Technical dept in this process usually very time-consuming risky and stressful.  
+
+Without testing infrastructure team can’t create the tests. It’s not necessary to became crazy and move all resources to create tests for existing code, but developers have to has possibility to creating the tests when they can and run in a regular bases. Without tests developers can’t effectively refactor code or implement a new features without introducing new bugs, so they should do all manual testing which is very ineffective, time-consuming and bad for team morale. Having this infrastructure will give the opportunity to became project health better over the time. Without it this will likely never happen.
+
+So, fixing the infrastructure issues is the key to manage the technical debt.  
+
+Data backing - it a most important part to decrease the risk. There is no way to make a lot of changes without mistakes, so team should not fear that their mistakes will result in data loss or corruption, which is, probably, the worst thing that can happen. 
+Managing the TD in code is more straightforward. Some solutions already covered by the best practices of used languages, tools and frameworks. To identify the problem and engineer the solution are not very difficult, and with the proper testing risk can be minimal. Refactoring, test creation and code reviews should be permanent part of the development process.  
+One more thing, which, I believe,  is very important to successful reduce of TD. Technical debt does not grow overnight. Sometimes it takes a years before people start thinking about to do something with, and it’s usually because they didn’t saw the whole picture to the very end, and start looking at it when any change became taking weeks and bugs and complaints numbers growth exponentially. This often because of issues organization for tech team. Leads organize bugs/stories/requests/tasks independently into different lists/projects, which hide the real scope of work. To avoid this, team should collect all technical tasks, bugs and issues in the one list. It will be a big list, but it will show how are you really deal with existing Technical Debt and implement features stories in the same time. Keeping things in different lists/boards/projects only hide the real situation. Splitting them to the different project has sense only when the separate teams will work on them.  
+
+So, to summarize all of that:
+
+1. Make user data safe
+2. Isolate and unify environment and configurations to avoid confusions
+3. Add tests infrastructure
+4. Make development more safe by adding high-level tests
+5. Make delivery process as simple as possible
+6. Cover changed code with tests
+7. Make real debt visible to everyone
+8. Use simple rules to make things clear  
+
+Hope it helps!
```

Comment:
It's, probably, structure. I'll think how to restructure. 
Yeah I thought about whese measurment, but they are not very helpful on project with the big TD, these tools are mostly about to prevention TD. They are useful when big isues are already fixed, but you're right, I should mention it.